app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/viewmodel/VideosInSameCategoryViewModel.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package android.support.v17.leanback.supportleanbackshowcase.app.room.viewmodel;
+
+import android.app.Application;
+import android.arch.lifecycle.AndroidViewModel;
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MediatorLiveData;
+import android.arch.lifecycle.Observer;
+import android.arch.lifecycle.ViewModel;
+import android.arch.lifecycle.ViewModelProvider;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.config.AppConfiguration;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.db.DatabaseHelper;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.db.entity.VideoEntity;
+
+import java.util.List;
+
+public class VideosInSameCategoryViewModel extends AndroidViewModel {
+
+    // The parameter used to create view model
+    private final String mCategory;
+
+    private DatabaseHelper mDatabaseHelper;
+
+    /**
+     * List of VideoEntities in same category
+     */
+    private LiveData<List<VideoEntity>> mVideosInSameCategory;
+
+
+    public VideosInSameCategoryViewModel(@NonNull Application application, final String category) {
+        super(application);
+        this.mCategory = category;
+
+        mDatabaseHelper = DatabaseHelper.getInstance();
+
+        // create database in main thread
+        mDatabaseHelper.createDb(this.getApplication());
```

Comment:
asking the VM to create the db is bad abstraction.
DatabaseHelper should just create it at first call.

Reply:
Thank you for your suggestion, the database creation logic is handled by the database helper now.